### ISSUE TITLE
Resolution Problems and random appearence of centered cursor.

### ISSUE THREAD

**Issue by 8FordPrefect8** [xbmc/xbmc#14977]
### Describe the bug
I am running LibreElec with Milhouse Builds on a generic x86 PC. Since the Milhouse build 1028 I have two annoying issues which are related I assume. First thing is related to screen refresh rate I guess. Starting a playback takes now a lot longer and is connected to a longer black screen. Also starting DVB Playback which is 50Hz here is now bugged. It always starts with a longer black phase then the system tries out 25Hz for a second and then it switches to 50HZ with a black phase in between.

With switching on and off Playback and therefore the resolution comes a second problem. With some chance after switching display mode a small cursor appears in the center of the screen. To get rid of it I have to start and stop playback several times. It is not the cursor of Kodi. Going to settings and activating the cursor there causes the appearence of a second unrelated kodi cursor.

I contacted Milhouse in the forum and he suggested that I make a bug report here, since from his perspective it has to be a kodi bug.
Here is the initial description:
https://forum.kodi.tv/showthread.php?tid=298462&pid=2793498#pid2793498
and here is Milhouses interpretation.
https://forum.kodi.tv/showthread.php?tid=298462&pid=2796470#pid2796470

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Start playback and get smooth and rapid transition. No cursors should appear anywhere.

## Actual Behavior

Starting playback results in short black screen before playback starts. With DVB Playback the wrong display mode (25Hz) is set in a first step. Leading to a several second annoying switching channel time. Start and stopping playback can trigger the appearance of an small cursor in the center of the screen.

### To Reproduce

1. Start system with Milhouse LibreElec later then 1027
2. Start Playback

### Debuglog

The debuglog can be found here:
https://paste.ubuntu.com/p/VYXt9cQH9t/

### Screenshots 
Here are some links or screenshots to help explain the problem:

https://i.imgur.com/UbAF1SH.jpg

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [x ] Linux
 - [ ] OSX
 - [ ] Raspberri-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by FernetMenta** [comment]
Whitelisting 25Hz refresh rate is a bad idea when using PVR. Either blame tvheadend for reporting wrong fps or remove 25hz from whitelist.

**Comment by GuillaumeDua** [comment]
I Have the same problem is centered cursor.

I use Fedora Workstation (latest version). When using Gnome as graphical environment, then launching Kodi as an application, no issue.

When starting on Kodi as graphical environment, a centered cursor is always on top.

Disabling cursor on Kodi's configuration "Input" section only remove the Kodi cursor (when using a mouse as input device).
Seems to be Fedora's mouse.
Existing fix on the web suggest to replace the OS mouse cursor by a 1x1 transparent pixel. Not a satisfying solution to me.

Setup : Latest Fedora workstation, latest Kodia available on fedora rpm.

**Comment by kib** [comment]
That really does not seem to be a related bug. @GuillaumeDua I suggest you open a thread at the Kodi forum for some input on what could be happening there.

**Comment by 8FordPrefect8** [comment]
I didn't whitelist or block any resolution. Actually I see no option for that. I would also not like to block 25Hz. I think I have some material with that and it worked fine before.

Regarding the Problem from @GuillaumeDua : I think they are related. My feeling is too that the cursor is the libreelec cursor which is also sometimes visible on bootup. My guess is that during this frequent frequency switching with short black screens the overlay is not working correctly anymore. But I have no clue how any of this works to be honest.

**Comment by 8FordPrefect8** [comment]
Okay, I once had a different custom build installed, back then when Milhouses Builds didn't support Gemini Lake. On this other build existed an option where I could white and black list frequencies. I had to whitelist 25 Hz because for example the movie Der Himmel über Berlin wouldn't work fine without it.

But more important. Please don't focus too much on the 25Hz issue. The main issue would persist even if I find the file in which I can black list 25Hz again. The behavior for every playback has changed for me. With version 1027 from Milhouse I start any movie and it will directly start. With 1028 everytime I will get a black screen before which seems like the display is lost for some short time.
And this loss of display connection is also what provokes the appearing of the cursor. I think the cursor is all the time in the background but when you often loose the display I have the feeling that there is a chance that the stacking of display elements gets mixed and the linux curor pops in the forground.

**Comment by FernetMenta** [comment]
A single bug report should only reference a single issue, not multiple ones. The issue of having multiple ones.

>  It always starts with a longer black phase then the system tries out 25Hz for a second and then it switches to 50HZ with a black phase in between.

this is solved by creating a whitelist not containing 25hz. 

I suggest that you first apply the resolution for this issue . If you encounter further issues after having applied this resolution, please open a new issue.

**Comment by 8FordPrefect8** [comment]
You are right, I should have splitted this. However the 25Hz was just an example. The general problem I tried to report here is that since this version I have problems with frequency changes at all frequencies.

**Comment by 8FordPrefect8** [comment]
I created a second log with 25Hz disabled.
https://paste.ubuntu.com/p/rnjrgTdCkC/

So lets really ignore 25Hz for the moment. The problem is that hdmi connection gets some trouble when playback is started. This leads in addition to the overlay of the libreelec cursor which is normally in the background, as far as I understood.

**Comment by MilhouseVH** [comment]
Just add that yes, the "appearance of centered mouse cursor" is probably a red herring - if you shut down Kodi on LibreELEC x86_64 you'll see exactly that, a centered mouse cursor as it's the default appearance for the underlying Fluxbox window manager. Best guess is that due to resolution/refresh rate changes, Kodi is losing/removes all the window layers temporarily, and/or long enough to expose the underlying Fluxbox framebuffer+mouse pointer (which you were not seeing before the whitelist change).

**Comment by 8FordPrefect8** [comment]
Thx, for confirmation.
So the main problem is indeed the changes in kodi how the switching of resolutions is handled.

**Comment by MilhouseVH** [comment]
Yes, almost certainly.

**Comment by FernetMenta** [comment]
The way resolution switches are handled for X11 hasn't changed for a long time. It was mentioned that the issue started Oct. 28th. If this is true, I suggest to revert https://github.com/xbmc/xbmc/pull/14727 and try again

**Comment by 8FordPrefect8** [comment]
Yep Milhouse build from 1027 is fine. With 1028 the problem started. I can switch back and forth but this is the line. I don't know how to revert this but I am of course willing to support with testing.

**Comment by MilhouseVH** [comment]
@8FordPrefect8 try this LibreELEC Generic build:

http://milhouse.libreelec.tv/builds/testing/Generic/LibreELEC-Generic.x86_64-9.0-Milhouse-20181207220942-%231207j-g6db1ac5.tar

It is based on latest Kodi 0c2d5431050e1cf3bfa2139bfc0644049a60127b with #14727 reverted.

**Comment by 8FordPrefect8** [comment]
Stupid question: I have to place this tar file in /storage/update and reboot right?

**Comment by FernetMenta** [comment]
looks like the latest change https://github.com/xbmc/xbmc/pull/15010 killed PVR in general. @MilhouseVH can you drop this too?

**Comment by MilhouseVH** [comment]
@8FordPrefect8 please test:

http://milhouse.libreelec.tv/builds/testing/Generic/LibreELEC-Generic.x86_64-9.0-Milhouse-20181208140410-%231207c-g6db1ac5.tar

This is 8cdec8f115777e9f43a81f9a5e2426e9e66d7313 with both #14727 and #15010 reverted.

**Comment by 8FordPrefect8** [comment]
Bug is still present. DVR is working again.
During last Playback I had the cursor and also switching is not working smoothly.
https://paste.ubuntu.com/p/47XgzJVGCM/

I didn't change the whitelisted resolutions again, but I assume that shouldn't matter.

**Comment by 8FordPrefect8** [comment]
And I am a bit trapped now. With your newest build I lost your update channel. Do I have to reconfigure it?

**Comment by FernetMenta** [comment]
@8FordPrefect8 your logs are impossible to read because full of

```
15:48:36.240 T:139858286733056   DEBUG: CAddonSettings[plugin.video.emby]: failed to find definition for setting LastIncrementalSync. Creating a setting on-the-fly...
15:48:36.240 T:139858286733056   DEBUG: CSettingsManager: requested setting (MinimumSetup) was not found.
15:48:36.240 T:139858286733056   DEBUG: CAddonSettings[plugin.video.emby]: failed to find definition for setting MinimumSetup. Creating a setting on-the-fly...
```

can you disable this addon and post a new log?
apart from the log spam I won't be surprised if this logging does any harm. the frequency of logging is < 1ms

**Comment by 8FordPrefect8** [comment]
I am sorry. This Emby for Kodi addon is really creazy.
This should be better. The cursor appeared finally during netflix playback.
https://paste.ubuntu.com/p/R3ZdzKDB5p/

**Comment by 8FordPrefect8** [comment]
Let me try to explain. I had once a completely different build when Milhouses was not working with my hardware. There I had a whitelist. Then when I switched back to Mihlouse there was not the need for this and I hadn't. Then in the last days on your request I made a whitelist to exclude the 25Hz.

**Comment by FernetMenta** [comment]
Well, if you hadn't had a whitelist before Oct 28th, you wouldn't had any switches of resolutions at all. Can you install 1027 again and verify that all is ok?

**Comment by 8FordPrefect8** [comment]
Ok, I will try to figure out how to go back to 1027 and do again a log with emby deactivated.
In the mean time: Lets be really specific about the whitelist. I am talking about going to settings/system/display (dont know the english label) and select there in expert mode allowed resolutions and frequencies. This was not set the complete last months and I had no problem at all with all the different sources. Everything was displayed correctly without any stutter. In case there could be some other config file under .kodi it would be helpful to know the location. Than I can check with nano whats in their.

I will post a new log as soon as I am ready.

**Comment by 8FordPrefect8** [comment]
Interesting. I started some files and it didn't worked with 1027. I then removed everything from the whitelist and it worked again immediately.
https://paste.ubuntu.com/p/27KjG95vvC/

**Comment by FernetMenta** [comment]
> I then removed everything from the whitelist and it worked again immediately.

The result is that Kodi does not change resolution anymore. If this is what you wanted to achieve, use current version, go to settings-player, and disable "adjust refresh rate to match video"

**Comment by 8FordPrefect8** [comment]
Nope thats not the solution. Of course if I disable this. I don't have any switches of the display during playback but I get lots of micro stutters if its not fitting.
With 1027 that was not the case.

**Comment by 8FordPrefect8** [comment]
With 1208 I don't have the 25 Hz issue with DVB anymore, even without a whitelist. However, the problems with the resolution switching and therefor the cursor are still present.

**Comment by 8FordPrefect8** [comment]
I am using a workaround now against the cursor by deactivating the cursor in the xserver startup.
This is not a fix, but at least it buggers me not anymore. Here is a link, how to do it in libreelec. https://forum.kodi.tv/showthread.php?tid=298462&pid=2798878#pid2798878
For  @GuillaumeDua it is not a solution as he of course wants to have a cursor in Fedora.

Do you think it would help if I provide a log with 1027 again without the whitelist and then with 1028 with exact same settings. FernetMenta had the idea I just have to deactivate resolution syncing and it would be the same as in 1027 without whitelist. But I can't believe it. Because I didn't notice any stuttering beforehand. I mean I can check again, but I would be really surprised. I can check if my display gives some note about the reolution it's running with.

**Comment by 8FordPrefect8** [comment]
I will be not present the next days or at least not in reach of my kodi installation, so I decided to create two logs in the morning.
Here is 1027 https://paste.kodi.tv/qobukupuxa.kodi.
And here is 1208 https://paste.kodi.tv/balipujoso.kodi.

I also had a look at the TVpanel information and indeed you were right Fernetmenta. Although in 1027 syncing of Frequencies at startup was set the frequency was all the time 60Hz. I am really surprised I didn't noticed this all the time. But after you suggested to switch off syncing I had a crtitical view on it and saw it. At least I will never need a 144Hz Display ^^.
That being said. Maybe the longer black screen is really just the laggy switching time of my display. I dumped once the display information because I got screen losses but I don't think that makes it worse. It was also a kernel bug for the intel GPU that most likely is fixed now.
The only thing since the 25Hz issue was fixed in parallel which remains is the cursor bug.
As somebody else has this problem too on a completely different distro, I think it's really a kodi bug.
Maybe one should create a new bug report and close this one. I would also assume that the bug was also present before 1027 because I had this issue beforehand and it dissappeared with switching to milhouse builds again. But I think it was only hidden, because now display mode switching occured anymore.
